app/controllers/repair_requests_controller.rb

```diff
@@ -1,57 +1,39 @@
 class RepairRequestsController < ApplicationController
   before_action :set_property
+  before_action :set_cautionary_contact
+  before_action :create_and_set_keyfax_session
+  before_action :set_keyfax_result
 
   def new
-
-    if keyfax_session_started?
-      if should_get_keyfax_result?
-        @keyfax_result = Hackney::KeyfaxResult.find(params["guid"])
-        unless sor_code_valid?
-          flash.now[:error] = [@keyfax_result.problem_description]
-        end
-      else
-        flash.now[:error] = ["Keyfax session cancelled"]
-      end
-    end
-
-    @cautionary_contact = Hackney::CautionaryContact.find_by_property_reference(@property.reference)
-    @keyfax_session = Hackney::KeyfaxSession.create(current_page_url: new_property_repair_request_url(@property.reference))
-    cautionary_contact_alert_codes = @cautionary_contact.alert_codes.join("; ")
-
-    @repair_request = Hackney::RepairRequest.new(
-      contact_attributes: {},
-      work_orders_attributes: [{
-        sor_code: @keyfax_result&.sor_code,
-        }],
-      priority: @keyfax_result&.priority,
-      description: "#{cautionary_contact_alert_codes} #{@keyfax_result&.fault_text}"
+    @repair_request = new_repair_request(
+      contact_attributes: {
+      },
+      tasks: [
+        new_task(
+          sor_code: @keyfax_result&.sor_code,
+        )
+      ],
+      priority: @keyfax_result&.priority || "N",
+      description: "%{alerts} %{text}" % {
```

Comment:
here you probably meant `"#{alerts} #{text}"`

Reply:
It's correct, it calls `sprintf`:
https://ruby-doc.org/core-2.5.1/String.html#method-i-25

Reply:
The `String#%` calls `sprintf`, and the hash after it are the parameters to substitute. Works like the arguments for the `t()` method to substitute.